mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -0,0 +1,164 @@
+
+    public static final String SOURCE_ATTRIBUTE_NAME = "source";
+    public static final String DESTINATION_ATTRIBUTE_NAME = "destination";
+    public static final String DEFAULT_SOURCE_ATTRIBUTE_NAME = "icalendar";
+    public static final String DEFAULT_DESTINATION_ATTRIBUTE_NAME = "icalendarJson";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ICALToJsonAttribute.class);
+
+    private final ObjectMapper objectMapper;
+    private String sourceAttributeName;
+    private String destinationAttributeName;
+
+    public ICALToJsonAttribute() {
+        this.objectMapper = new ObjectMapper()
+            .registerModule(new Jdk8Module());
+    }
+
+    public String getSourceAttributeName() {
+        return sourceAttributeName;
+    }
+
+    public String getDestinationAttributeName() {
+        return destinationAttributeName;
+    }
+
+    @Override
+    public String getMailetInfo() {
+        return "ICALToJson Mailet";
+    }
+
+    @Override
+    public void init() throws MessagingException {
+        sourceAttributeName = getInitParameter(SOURCE_ATTRIBUTE_NAME, DEFAULT_SOURCE_ATTRIBUTE_NAME);
+        destinationAttributeName = getInitParameter(DESTINATION_ATTRIBUTE_NAME, DEFAULT_DESTINATION_ATTRIBUTE_NAME);
+        if (Strings.isNullOrEmpty(sourceAttributeName)) {
+            throw new MessagingException(SOURCE_ATTRIBUTE_NAME + " configuration parameter can not be null or empty");
+        }
+        if (Strings.isNullOrEmpty(destinationAttributeName)) {
+            throw new MessagingException(DESTINATION_ATTRIBUTE_NAME + " configuration parameter can not be null or empty");
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        if (mail.getAttribute(sourceAttributeName) == null) {
+            return;
+        }
+        if (mail.getSender() == null) {
+            return;
+        }
+        try {
+            Map<String, Calendar> calendars = (Map<String, Calendar>) mail.getAttribute(sourceAttributeName);
+            List<String> collect = calendars.values()
+                .stream()
+                .flatMap(calendar -> toJson(calendar, mail))
+                .collect(Guavate.toImmutableList());
+            mail.setAttribute(destinationAttributeName, (Serializable) collect);
+        } catch (ClassCastException e) {
+            LOGGER.error("Received a mail with {} not being an ICAL object for mail {}", e, sourceAttributeName, mail.getName());
+        }
+    }
+
+    private Stream<String> toJson(Calendar calendar, Mail mail) {
+        return mail.getRecipients()
+            .stream()
+            .map(recipient -> toJson(calendar, recipient, mail.getSender(), mail.getName()))
+            .filter(Optional::isPresent)
+            .map(Optional::get);
+    }
+
+    private Optional<String> toJson(Calendar calendar, MailAddress recipient, MailAddress sender, String mailName) {
```

Comment:
I'd rather split the object construction and the json serialization in two methods